Source/BusinessMomentsAdapters/Backend/domain/adapters/for_Adapter/when_adding_connector/and_there_is_an_existing_with_same_name.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { Adapter, ConnectorWithNameAlreadyExists } from '../../Adapter';
+import { EventSourceId } from '@dolittle/sdk.events';
+import { Guid } from '@dolittle/rudiments';
+import { ConnectorTypes } from '../../../../concepts';
+import { BrokenRuleError } from '@dolittle/vanir-backend';
+
+describe('when adding connector and there is an existing with same name', () => {
+    const name = 'Some Connector';
+    const adapter = new Adapter(EventSourceId.from('46e44f22-1e76-4110-8b1d-d088cd3a1481'));
+    adapter.onConnectorAddedToAdapter({
+        name,
+        connectorId: 'a60bd940-7a3f-417d-bf2d-8126513fe3fc',
+        connectorTypeId: ConnectorTypes.WebHook.toString()
+    });
+
+    let brokenRuleError: BrokenRuleError;
+
+    try {
```

Comment:
The variable is outside the beforeEach - but its value is set in the beforeEach. If we had the declaration inside the beforeEach, we wouldn't have access to it in the `it` statements

Reply:
@einari I agree regarding scoping the variable outside the beforeEach.

However, I can't see there is a `beforeEach` being invoked in this test? The way I interpret this code (in my head) is that adaptor.onConnectorAddedToAdapter() method invocation will be called just once - when the describe block is being invoked. If there was multiple it-blocks in this file the initialisation would only be executed once - hence the temporal coupling. If variables was initialised in an actual beforeEach the preconditions would have been the same for all the it-blocks.

Reply:
:) Sorry, I was completely out of context - and assumed there was a beforeEach..  

Good point about the temporal coupling.
I don't necessarily put in beforeEach for all specs. I don't think for this case there would be any benefit in having a beforeEach now. Maybe if I were adding another it - but that would be depend on what I was doing. 

One could argue we should always do it, to avoid potentially having problems. Which would be fine by me.